pypowerwall/tedapi/decorators.py

```diff
@@ -17,3 +19,57 @@ def wrapper(*args, **kwargs):
         return func(*args, **kwargs)
 
     return wrapper
+
+
+# Connection Decorator
+# Checks to see whether a connection exists, and if not, attempts to connect
+def uses_connection_required(func):
+    @functools.wraps(func)
+    def wrapper(self, *args, **kwargs):
+        if not self.din:
+            if not self.connect():
+                log.error("Not Connected - Unable to get status")
+                return None
+        return func(self, *args, **kwargs)
+    return wrapper
+
+
+# Cache Decorator
+# Checks to see whether a cached result exists for the function, and if so, returns it. If not, it calls the function and caches the result.
+def uses_cache(cache_key):
+    def decorator(func):
+        def wrapper(self, *args, **kwargs):
+            # We probably can't guarantee that "force" will be the first argument for the function, so let's find it
+            sig = inspect.signature(func)
+            namedArgs = sig.bind(self, *args, **kwargs).arguments
+            force = False
+            if 'force' in namedArgs:
```

Comment:
Are we concerned with case sensitivity here?

Reply:
Not really, to get there you'd have to have ignored the convention of the rest of the file to be the only capitalized argument name, and you'd know it pretty quick

Reply:
Sounds good.